Source/JavaScriptCore/jit/JITInlines.h

```diff
@@ -477,6 +477,44 @@ ALWAYS_INLINE ECMAMode JIT::ecmaMode<OpPutPrivateName>(OpPutPrivateName)
     return ECMAMode::strict();
 }
 
+template<size_t minAlign, typename Bytecode>
+ALWAYS_INLINE MacroAssembler::Address JIT::computeGoodBaseAddressForMetadata(const Bytecode& bytecode, GPRReg metadataGPR)
+{
+    // This function attempts to decide what the best base address is when
+    // loading fields from a bytecode instruction's metadata. If offsets are
+    // small enough, we want to emit a single load directly offset from the
+    // metadata table register. But if they're too big, we want to materialize
+    // the metadata offset of the current instruction into a register only
+    // once, so repeated loads don't need to redo that arithmetic.
+
+    // Note: minAlign is very important to get right, but hard to check for
+    // automatically. It should be the minimum alignment or access size across
+    // all the operations we use this address for. This is critical on ARM64
+    // (where this function is most useful) because we have a substantially
+    // larger addressible range when we can assume the access is aligned to
+    // the access size. One important note is that we assume that any accesses
+    // larger than the specified minAlign are also aligned - only then is it
+    // the case that an offset being encodable at a smaller access size also
+    // implies that it is encodable at a larger access size.
+
+    using Metadata = typename Bytecode::Metadata;
+    static_assert(alignof(Metadata) >= minAlign);
+    uint32_t metadataOffset = m_profiledCodeBlock->metadataTable()->offsetInMetadataTable(bytecode);
+    bool shouldEncodeMetadataOffsetInline = metadataOffset < 1 << 30;
```

Comment:
Let's use `(1 << 30)` to be more explicit about this.

Are you sure that that `metadataOffset < 1 << 30` is the right condition for all non-ARM64 CPUs?  If you're not certain, the better thing would be to just disable this optimization for those CPUs (i.e. `shouldEncodeMetadataOffsetInline == false`), right?

Reply:
It may not strictly be ideal for every platform, although any all-or-nothing change is going to be imperfect. I think this is probably the right heuristic for x86_64 at least (since 32-bit displacements are fine to encode in a single instruction), to minimize friction it would probably be better to disable this on ARMv7/RISC-V since that'll pretty much just give the current behavior. I'll make that change in the next push.